### ISSUE TITLE
with-react-native-web doesn't work with last version of next.js

### ISSUE THREAD

**Issue by AMSTKO** [vercel/next.js#31255]
### What example does this report relate to?

with-react-native-web

### What version of Next.js are you using?

12.0.3

### What version of Node.js are you using?

16.13.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next

### Describe the Bug

Since next@12.0.3 the With React Native Web example doesn't work, it's fine with next@12.0.2

```
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
TypeError: Class constructor MyDocument cannot be invoked without 'new'
```

### Expected Behavior

"With React Native Web" Example functional with last version of next.js.

### To Reproduce

Try to start this example https://github.com/vercel/next.js/tree/master/examples/with-react-native-web

**Comment by efoken** [comment]
Havig the same issue in an React Native Web project with Next.js 12. I think it's because there is a new `document.web.js` file in the `next/document` package, which get's imported because of the adapted webpack config:

```js
module.exports = {
  webpack: (config) => {
    config.resolve.alias = {
      ...(config.resolve.alias || {}),
      // Transform all direct `react-native` imports to `react-native-web`
      'react-native$': 'react-native-web',
    }
    // The following lines break _document.js:
    // config.resolve.extensions = [
    //   '.web.js',
    //   '.web.ts',
    //   '.web.tsx',
    //   ...config.resolve.extensions,
    // ]
    return config
  },
}
```

Commenting out `config.resolve.extensions` works, but it breaks many common third-party React Native packages...

**Comment by balazsorban44** [comment]
Looks like this was fixed (probably in #32076?) because I could not reproduce any errors anymore.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
